modules/versions.xqm

```diff
@@ -39,13 +39,25 @@ declare function versions:find-compatible-packages(
             $packages[version = $version]
         else if ($semver-min and $semver-max) then
             versions:find-version($packages, $semver-min, $semver-max)
-        else if 
-            (
-                $exist-version-semver and
-                versions:is-newer-or-same($exist-version-semver, $package/requires/@semver-min) and
-                versions:is-older-or-same($exist-version-semver, $package/requires/@semver-max)
-            ) then
-            $package
+        else if (exists($exist-version-semver)) then
+            if (
```

Comment:
if-statements inside an if-condition is hard to read and understand. Can this be refactored in one of the following ways?

```xquery
(
  not(exists($package/requires/@semver-min)) or 
  semver:ge($exist-version-semver, $package/requires/@semver-min, true())
) and
(
  not(exists($package/requires/@semver-max)) or
  semver:lt($exist-version-semver, $package/requires/@semver-max, true())
)
```
or create a function that is called as in the case above with `versions:find-version`

Reply:
Great idea, thanks. I will do so.

Reply:
Ah, I remember why I needed the conditional. Because if the `@semver-min` attribute is not present, then `semver:ge` will throw an error. I'll work on an alternative.

Reply:
Ok, I created a function; hopefully this is improved? See [Simplify checking of eXist version dependencies](https://github.com/eXist-db/public-repo/pull/77/commits/0791816a8a077d853349bd25cc44c7157464885f). 

p.s. In the fix to #72 I will clean up the profusion of under-documented and similarly named functions in this module.